bioconda_utils/lint_functions.py

```diff
@@ -185,6 +203,22 @@ def has_windows_bat_file(recipe, meta, df):
             'fix': 'remove windows .bat files'
         }
 
+
+def should_be_noarch(recipe, meta, df):
+    deps = _get_deps(meta)
+    if (
+        ('gcc' not in deps) and
+        ('python' in deps) and
+        not _has_preprocessing_selector(meta)
```

Comment:
I'm worried that this disables the linter for too many packages. I think there are a few packages with pre-processing selectors in the `url` section or test section - still we would like to have noarch packages for it. Just a feeling, not sure how often these cases are.